src/test/kotlin/org/pets/history/controller/PetControllerTest.kt

```diff
@@ -218,18 +229,58 @@ class PetControllerTest {
         val id = pet.id
 
         mockMvc.perform(
-            get("/pets/$id/medical-records")
-                .accept(MediaType.APPLICATION_JSON)
-                .contentType(MediaType.APPLICATION_JSON)
+                get("/pets/$id/medical-records")
+                        .accept(MediaType.APPLICATION_JSON)
+                        .contentType(MediaType.APPLICATION_JSON)
+        )
+                .andExpect(status().isOk)
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.results.length()").value(1))
+                .andExpect(jsonPath("$.results.[0].id").isNumber)
+                .andExpect(jsonPath("$.results.[0].address").value("Evergreen 123"))
+                .andExpect(jsonPath("$.results.[0].datetime").value("2024-03-28T12:00:00"))
+                .andExpect(jsonPath("$.results.[0].specialist").value("Lisa Simpson"))
+                .andExpect(jsonPath("$.results.[0].observations").value("Healthy"))
+    }
+
+    @Test
+    fun `Given a pet id that does exist and the details of a treatment, registers said record and return it`() {
+        val pet = petRepository.save(anyPet())
+        val id = pet.id
+
+        val treatment = aTreatment()
+        val json = mapper.writeValueAsString(treatment)
+
+        mockMvc.perform(
+                post("/pets/$id/treatment")

```

Comment:
Efecto de la sugerencia anterior.

```suggestion
                post("/pets/$id/treatments")

```